tools/version_check.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+# For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+set -euo pipefail
+#set -o posix;
+LC_CTYPE="C"
+
+# WARNING: depends on word splitting
+semver2num () { printf "%03.0f%03.0f%03.0f" $(echo "$1" | sed -e "s/[^[:digit:]]*\([[:digit:]]\+\)\.\([[:digit:]]\+\)\.\([[:digit:]]\+\).*/\\1 \\2 \\3/"); }
```

Comment:
Yes, I appreciate the note and I did take it to heart as it was the inspiration for making this one-liner.  However, the only thing it did was extract into an array but making them comparable requires a step similar to the printf code.  This code turns the semantic version numbers into a single number which can be directly compared.  On top of that, it's POSIX compliant which means if some random person grabs this code, they won't be disappointed when it doesn't work on their embedded system's tiny shell.

Reply:
@GravisZro it really doesn't - you can just check elements from the array read. And no additional tooling is required so actually it's better by your own argumentation :) 
Saying that you want to run clang on an embedded system tiny shell? 
If it's capable to run clang, then it's more than capable to call bash... 
Also, what do you really mean when you say: "It's POSIX compliant?"

Reply:
> you can just check elements from the array read

That means doing a long comparison of all three elements like the one I had written before.

>  Saying that you want to run clang on an embedded system tiny shell?

No, I'm saying copying lines of the script will work anywhere.  If someone wanted to easy semantic version comparison function then it would work on their shell, even if it's not Bash.

>  Also, what do you really mean when you say: "It's POSIX compliant?"

POSIX is a specification for UNIX.  What this means is that if write something that is POSIX complaint that **will** run on anything that complies with POSIX specification.  In reality, this means my script will run on _any_ UNIX-like shell, far beyond just the GNU userland tools.  Here's the spec: https://pubs.opengroup.org/onlinepubs/9699919799/